extensions/spark/kyuubi-spark-jvm-quake/src/main/scala/org/apache/spark/kyuubi/jvm/quake/SparkJVMQuake.scala

```diff
@@ -0,0 +1,139 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.kyuubi.jvm.quake
+
+import java.io.File
+import java.lang.management.ManagementFactory
+import java.util.concurrent.TimeUnit
+
+import scala.collection.JavaConverters._
+
+import SparkJVMQuake._
+import com.sun.management.HotSpotDiagnosticMXBean
+import org.apache.spark.SparkConf
+import org.apache.spark.internal.Logging
+import org.apache.spark.kyuubi.jvm.quake.SparkJVMQuakeConf._
+import org.apache.spark.util.ThreadUtils
+
+class SparkJVMQuake(conf: SparkConf, heapDumpEnabled: Boolean) extends Logging {
+  private val appId = conf.get("spark.app.id", "app-id").stripPrefix("application_")
```

Comment:
why strip "application_"? the app id could be different patterns, e.g. on local mode, it's `local-xxx`; on k8s mode, it's `spark-xxx`

Reply:
the strip is deleted